app/frontend/actions.ts

```diff
@@ -121,52 +120,43 @@ export default ({setState, getState}: {setState: SetStateFn; getState: GetStateF
 
   /* LOADING WALLET */
 
-  const loadWallet = async (state, {cryptoProviderType, walletSecretDef, isWebUSB}) => {
-    // loadingAction(state, `Waiting for ${state.hwWalletName}...`)
+  const loadWallet = async (
+    state: State,
+    {cryptoProviderType, walletSecretDef, isWebUSB, shouldExportPubKeyBulk}
+  ) => {
     loadingAction(state, 'Loading wallet data...', {
       walletLoadingError: undefined,
     })
     const isShelleyCompatible = !(walletSecretDef && walletSecretDef.derivationScheme.type === 'v1')
+    const config = {...ADALITE_CONFIG, isShelleyCompatible, shouldExportPubKeyBulk}
     try {
-      switch (ADALITE_CONFIG.ADALITE_CARDANO_VERSION) {
-        case 'byron': {
-          const cryptoProvider = await CryptoProviderFactory.getCryptoProvider(cryptoProviderType, {
-            walletSecretDef,
-            network: NETWORKS.BYRON.MAINNET,
-            config: ADALITE_CONFIG,
-          })
-
-          wallet = CardanoWallet({
-            cryptoProvider,
-            config: ADALITE_CONFIG,
-          })
-          break
-        }
-        case 'shelley': {
-          const cryptoProvider = await ShelleyCryptoProviderFactory.getCryptoProvider(
-            cryptoProviderType,
-            {
-              walletSecretDef,
-              network: NETWORKS.SHELLEY[ADALITE_CONFIG.ADALITE_NETWORK],
-              config: ADALITE_CONFIG,
-              isWebUSB,
-            }
-          )
-
-          wallet = await ShelleyWallet({
-            config: ADALITE_CONFIG,
-            cryptoProvider,
-            isShelleyCompatible,
-          })
-          break
+      const cryptoProvider = await ShelleyCryptoProviderFactory.getCryptoProvider(
+        cryptoProviderType,
+        {
+          walletSecretDef,
+          network: NETWORKS.SHELLEY[ADALITE_CONFIG.ADALITE_NETWORK],
+          config,
+          isWebUSB, // TODO: into config
```

Comment:
worth rebasing to latest develop which refactors this to "forceWebUSB"

Reply:
I will, sorry.